src/main/java/bio/terra/workspace/app/configuration/DataRepoConfig.java

```diff
@@ -7,7 +7,7 @@
 
 @Configuration
 @EnableConfigurationProperties
-@ConfigurationProperties(prefix = "datarepo")
+@ConfigurationProperties(prefix = "workspace.datarepo")
 public class DataRepoConfig {
   private HashMap<String, String> instances;
```

Comment:
[aside / nit] I don't love the idea that we have an untyped key-value hash being specified in config like this. It just generally goes against the grain of our pattern "configs are mapped to FooConfig classes with typed value bindings".

At the very least – one convention I'd like to align on is that the configuration Java class is the canonical place where config values are documented (see [this section](https://docs.google.com/document/d/1VfgL8D6PpJg7irbL_gKnmyZ5_GH5OYGfK8GPN45odUs/edit#heading=h.83dycur23l5f)). Maybe doesn't need to be in this PR, but it would be great to see a descriptive comment added to the field definition for every value that's getting bound to a Spring app config.

Reply:
I'm curious what would be better than a `HashMap<String, String>` in this particular instance. We could potentially make this a little more strongly typed as `HashMap<String, URL>` and/or we could create a bespoke model class to represent the keys, something like `HashMap<DataRepoName, URL>`. Would that be better?

For this particular config value, we use a name to refer to each data repo instance, to allow the repo instance URLs to change without having to change any persisted data that references that instance. We *could* hardcode the keys (i.e. the names), but that also feels wrong.

Totally agreed that a code comment would help.

Reply:
What if we add metadata? https://docs.spring.io/spring-boot/docs/current/reference/html/appendix-configuration-metadata.htm

Reply:
Also, I'm going to adhere to that doc you linked Greg, very nice

Reply:
I guess I was more wondering aloud about why this property needs to support arbitrary keys. I can think of ~two scenarios:

(1) Once we can specify each per-environment TDR target more easily (which this PR starts to enable via profiles), we no longer need to specify additional named TDR targets, e.g. there's only one. So this can become a singular property `workspace.datarepo.instance`.

(2) We still want to support differing numbers / sets of TDR targets. If this is the case, I would question whether a simple key-value will be flexible enough in the long term. For example, is it plausible that we might need to track more complex metadata about each target TDR? I might be inclined to treat this config property as a single JSON-serialized string that maps to a list of objects with a known POJO structure. For example:

```
--- 
workspace: 
  datarepo: 
    # Instances is a multiline string that serializes to a list of 
    instances: |-
        [
         {"name": "azure", "url": "https://data.terra-azure.bio", "apiVersion": 2}
        ]
    primary-instance: "https://data.terra.bio"
```